x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -62,7 +63,12 @@ protected IndicesAccessControl(IndicesAccessControl copy) {
     @Nullable
     public IndexAccessControl getIndexPermissions(String index) {
         Tuple<String, String> indexAndSelector = IndexNameExpressionResolver.splitSelectorExpression(index);
-        return this.getAllIndexPermissions().get(indexAndSelector.v1());
+        return this.getAllIndexPermissions()
```

Comment:
This bit I'm not 100% confident on -- generally speaking, `getIndexPermissions` should be used for FLS/DLS which we are opting not to support in the first place, so all "valid" access should happen without selectors, anyhow (we only include the `::failures` in the map key, and omit it for data -- see also [here](https://github.com/elastic/elasticsearch/pull/123986/files#diff-29b4a5438dc9bdb4c266e5d6818869e494af86cad6640d8c236ca4389090cd0cR558)).

What I need to convince myself of still is that this will fail gracefully. I didn't want to block an initial high-level review on this though.